src/application/communication/GameClient.java

```diff
@@ -163,7 +185,13 @@ private void joinGame(Pane gameBox, Specializations spec)
         System.out.println("trying to join game");
         try
         {
-            Broker broker = new Broker();
+            if(broker != null)
+            {
+                broker.unregisterAll(this);
```

Comment:
unregisterAll, unregisters gameclient as a observer from all observations right? There is no deactivate in gameclient, so what deactivate are you talking about?